src/views/QueryEditor/components/QueryBuilder/hooks/useConnectionData.ts

```diff
@@ -74,6 +74,19 @@ export const useConnectionData = (query, datasource) => {
             "type = 'UInt32' " +
             'ORDER BY name';
           break;
+        case TimestampFormat.Float:
+          query =
+            'SELECT name ' +
+            'FROM system.columns ' +
+            "WHERE database = '" +
+            selectedDatabase +
+            "' AND " +
+            "table = '" +
+            selectedTable +
+            "' AND " +
+            "type LIKE 'Float%' OR type LIKE 'Decimal%' " +
+            'ORDER BY name';
+          break;
```

Comment:
great let's add separate TimestampFormat.Timestamp64_X, 

and use following query
TimeStamp(3)
TimeStamp(6)
TimeStamp(9)

`SELECT name FROM system.columns WHERE type LIKE 'UInt%' OR type LIKE 'Int%' WHERE database=? AND table=?`